handlers/common.go

```diff
@@ -31,10 +32,58 @@ func PingNoOp(event *events.Event, apiClient *client.RancherClient) error {
 	return nil
 }
 
+func acquire_networked_global_persistent_lock(resourceId string) error {
+	return nil
+}
+
+func release_lock(lock string) {
+	return
+}
+
 func getMachineDir(machine *client.Machine) (string, error) {
 	return getBaseMachineDir(machine.ExternalId)
 }
 
+func sync_resource(machine *client.Machine) error {
```

Comment:
@cjellick - Yeah, can think of a scenario where the files are out of sync (other than the partial failure case). If "create" happens on one node, and "bootstrap" happens on another, the "bootstrapped" file will not be present on the first node. This doesn't affect anything right now, but in the future, if we assumed that the folders were in sync across multiple gms', we'd be wrong. 

So, yeah, we can resync on every request. 

Also, there is another scenario I thought of while thinking about this. On "delete" of a host, we only remove the files from one of the gms'. The other gms' will continue to have these files. Assuming that we want to keep files in sync, do we want to clean up these deleted hosts too?